File: api/internal/infrastructure/persistence/game_charge.go
package persistence

type GameCharge struct {
	Base

	GameID       string  `gorm:"type:uuid;not null;index" json:"gameId"`
	UserID       *string `gorm:"type:uuid;index" json:"userId"`
	ExternalName string  `gorm:"type:varchar(255)" json:"externalName"`
	AmountCents  int     `gorm:"not null;check:amount_cents > 0" json:"amountCents"`

	Allocations []PaymentAllocation `gorm:"foreignKey:GameChargeID" json:"allocations"`
	Game        Game                `gorm:"foreignKey:GameID;constraint:OnDelete:CASCADE" json:"game"`
}

File: bouncy-web/src/lib/models/game-charge.ts
import type { Game } from './game';
import type { PaymentAllocation } from './payment-allocation';

export interface GameCharge {
  id: string;
  gameId: string;
  userId: string;
  externalName?: string;
  amountCents: number;
  createdAt: Date;
  allocations: PaymentAllocation[];
  game?: Game;
}
